### PR TITLE
chore(ingestion): move mmdb database in worker process

### PR DESCRIPTION
**Pull request by xvello**
## Problem

We need to improve the per-pod throughput of `plugin-server-ingestion`. While the kafka changes recently merged in `recordings-ingestion` should help us a lot, we're looking for more low-hanging fruits.

The app with the biggest footprint is by far the GeoIP app, as it runs for almost every event that we ingest. The lookups themselves are supposed to be fast, but the TCP talk between the main thread and the piscina worker look costly.

## Changes

- Bring the MMDB database into the Piscina worker, while still sharing it between pluginconfigs in that runner

We are currently operating with a single worker thread, as we're currently unable to saturate anyway (this could change once the kafka improvements are in). This PR should not have any memory impact.

If we were to increase the number of workers / pod in the future, we'll duplicate the memory cost of the MMDB database, as it's local to the worker.

## How did you test this code?

Test coverage + local devenv (patching capture.py to inject a french IP address in the event, instead of localhost):
image

### CODE REVIEW

File: plugin-server/src/worker/vm/extensions/geoip.ts [hunk 1]
```diff
@@ -1,12 +1,15 @@
 import { GeoIPExtension } from '@posthog/plugin-scaffold'
 
 import { Hub } from '../../../types'
-import { fetchIpLocationInternally } from '../../mmdb'
 
 export function createGeoIp(server: Hub): GeoIPExtension {
     return {
         locate: async function (ipAddress) {
```
Comment (xvello) [inline plugin-server/src/worker/vm/extensions/geoip.ts:7]: keeping the same signature for compatibility, although this function does not need to be async anymore

**Review by fuziontech — APPROVED** [review]
Ran through this and it looks great.

File: plugin-server/src/worker/vm/extensions/geoip.ts [hunk 2]
```diff
@@ -1,12 +1,20 @@
 import { GeoIPExtension } from '@posthog/plugin-scaffold'
 
 import { Hub } from '../../../types'
-import { fetchIpLocationInternally } from '../../mmdb'
```
Comment (hazzadous) [inline plugin-server/src/worker/vm/extensions/geoip.ts:4]: Maybe lost in the diff, but did you also remove `fetchIpLocationInternally` or is it still be being used elsewhere?

File: plugin-server/src/worker/vm/extensions/geoip.ts [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
 
 export function createGeoIp(server: Hub): GeoIPExtension {
     return {
         locate: async function (ipAddress) {
-            return await fetchIpLocationInternally(ipAddress, server)
+            if (server.mmdb) {
+                try {
+                    return Promise.resolve(server.mmdb.city(ipAddress))
+                } catch (e) {
```
Comment (hazzadous) [inline plugin-server/src/worker/vm/extensions/geoip.ts:11]: Should we be logging this errors. Maybe not as an error but a warning?

**Review by hazzadous — APPROVED** [review]
Looks good to me

**Pull request merged**